File: front-veterinaria/src/app/Services/banio.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Banio, BanioCustom, BanioMapper } from '../Models/banio';
import { HttpClient } from '@angular/common/http';
import { API_URL } from '../Environments/routes';

@Injectable({
  providedIn: 'root'
})
export class BanioService {

  constructor(private http: HttpClient) { }

  findAll(): Observable<Banio[]> {
    return this.http.get<Banio[]>(API_URL.banios.find_all);
  }

  findAllCustom(): Observable<BanioCustom[]> {
    return this.http.get<BanioCustom[]>(API_URL.banios.find_all_custom);
  }

  findAllMapper(): Observable<BanioMapper[]> {
    return this.http.get<BanioMapper[]>(API_URL.banios.find_all_mapper);
  }

  findAllByMascotaId(id:number): Observable<BanioMapper[]> {
    return this.http.get<BanioMapper[]>(API_URL.banios.find_all_by_mascota_id(id));
  }

  findById(id: number): Observable<Banio> {
    return this.http.get<Banio>(API_URL.banios.find_by_id(id));
  }

  insert(banio: Banio): Observable<Banio> {
    return this.http.post<Banio>(API_URL.banios.insert, banio);
  }

  update(id: number, banio: Banio): Observable<Banio> {
    return this.http.put<Banio>(API_URL.banios.update(id), banio);
  }

  delete(id: number): Observable<string> {
    return this.http.delete(API_URL.banios.delete(id), {responseType: 'text'});
  }
}

File: back-veterinaria/src/main/java/idat/proyecto/veterinaria/service/BanioService.java
package idat.proyecto.veterinaria.service;

import org.springframework.http.ResponseEntity;
import org.springframework.web.multipart.MultipartFile;

import idat.proyecto.veterinaria.entity.Banio;

public interface BanioService {
	public abstract ResponseEntity<?> insert(Banio banio);
	public abstract ResponseEntity<?> update(Integer id, Banio banio);
	public abstract ResponseEntity<?> delete(Integer id);
	public abstract ResponseEntity<?> findById(Integer id);
	public abstract ResponseEntity<?> findAll();
	public abstract ResponseEntity<?> findAllCustom();
	public abstract ResponseEntity<?> findAllMapper();
	public abstract ResponseEntity<?> findAllByMascotaId(Integer mascota_id);
	public abstract ResponseEntity<?> setFotoEntrada(Integer id, MultipartFile file);
	public abstract ResponseEntity<?> setFotoSalida(Integer id, MultipartFile file);
}
